### DISCUSSION TITLE
Quest 2 support

### DISCUSSION THREAD

**Discussion by soulicious1** [yasirkula/UnityNativeFilePicker#22] [General]
I have this problem when using the filepicker in an application build for VR (Meta Quest 2), it asks for permissions as it should be but then I am left with a loading spinner. It does log "OnApplicationFocus(false)" and "OnApplicationPause(true)". If I focus the app - by opening it again - I get a "files == null" (cancelled). Is it possible that this is not supported on Quest 2? Anyone who has some experience concerning this topic?
thanks in advance

**Comment by yasirkula** [comment]
Can you try all 4 combinations of the following values in Awake/Start:

```csharp
using( AndroidJavaClass ajc = new AndroidJavaClass( "com.yasirkula.unity.NativeFilePickerPickFragment" ) )
	ajc.SetStatic<int>( "pickerMode", 1 ); // 1 or 2

using( AndroidJavaClass ajc = new AndroidJavaClass( "com.yasirkula.unity.NativeFilePicker" ) )
	ajc.SetStatic<bool>( "UseDefaultFilePickerApp", true ); // true or false
```

If these don't work, then the used Android API might not be supported on VR. I unfortunately couldn't find any related info on the internet so I can't verify it. An alternative would be to use SimpleFileBrowser: https://github.com/yasirkula/UnitySimpleFileBrowser

**Reply by soulicious1** [reply]
These didn't work for me. Thanks again for providing help!

**Comment by fmbnicola** [comment]
Hey! I've also been trying to use this plugin for a Meta Quest 2 app. In my case, I'm able to get the read/write permission and when I call the file picker to select a pdf file (like the example provided on the readme file) I am able to select a file through the quest's native android file picker.

However, after doing this, it automatically returns to the app and never executes the callback function. OnFilePicked is never called.
Any clue what might be going on here or what I could try to do to fix this? Any help is greatly appreciated.

**Reply by yasirkula** [reply]
Hmm, wish I had a clue but not at the moment. I've no experience with VR. It's even a surprise to me that a file picker is displayed on the headset. You may also give [SimpleFileBrowser](https://github.com/yasirkula/UnitySimpleFileBrowser) a shot which uses uGUI. If Quest 2 uses Storage Access Framework, then this solution may not work either though.

**Reply by fmbnicola** [reply]
Thanks for the quick reply! As far as I understand it there isn't that much different about it being in VR, it is custom but at its core I believe it is running Android 10.

I actually started tackling this issue by using SimpleFileBrowser (another great plugin btw :D). 
It does work to an extent, I am able to browse the quests file system, see folders etc but for some reason it will not show pdf files, even when no file filter is added it just does not seem to be able to detect these files.

I had it working at some point but I think there might have been a software update in the meantime, or maybe I changed something in the unity Android settings, I'm not sure. Do you know of anything that could be causing this?

**Reply by yasirkula** [reply]
You might be picking folders instead of files. Otherwise, a fresh reinstall of the plugin might help if you had modified the plugin.

**Reply by fmbnicola** [reply]
I am definitely picking files: FileBrowser.WaitForLoadDialog(FileBrowser.PickMode.Files, false, null, null, "Select File", "Load"); 
I've done a little more testing and the results I am seeing are really weird, I am able to sometimes see other file types like .png and .mp4 but never .pdf and sometimes the file browser will just stop detecting even those files.

Unfortunately no, I did not modify the plugin. I will try updating to the latest version anwyay :(

**Reply by yasirkula** [reply]
Are any exceptions logged to logcat/console?

**Reply by fmbnicola** [reply]
Sorry I took a while. Thanks for suggesting checking logcat, although I did not find any exceptions while using the simple file browser plugin, when trying the native file picker solution I noticed that the callback for the file selection actually is running but only after I lock and then unlock the device... 

I could see Unity firing "Selected media uri: content: ..." message and the debug.log message I added with the correct path. But this only shows up on logcat after unlocking the quest.... Any clue what could cause something like this?

**Reply by yasirkula** [reply]
Wish I had. I think I've heard about the same issue before in a different native app (an Android wallpaper app). I'll conclude that the plugin isn't compatible with Quest because I don't have resources to investigate the issue and I don't even know where to start :/

**Reply by fmbnicola** [reply]
Yah that's understandable. Really frustrating that it seems so close to working but just out of reach :( Thanks for the help anyway!
